src/screens/IndexScreen.js

```diff
@@ -367,15 +390,15 @@ export const IndexScreen = ({ navigation, route }) => {
             const initialNewsItemsFetch =
               query === QUERY_TYPES.NEWS_ITEMS &&
               !Object.prototype.hasOwnProperty.call(queryVariables, [
-                keyForSelectedValueByQuery?.[query]
+                keyForSelectedValueByQuery(query)
               ]) &&
               !Object.prototype.hasOwnProperty.call(queryVariables, 'refetch');
 
             // apply additional data if volunteer events should be presented and
             // no category selection is made, because the category has nothing to do with
             // volunteer data
             const additionalData =
-              isCalendarWithVolunteerEvents && !hasCategoryFilterSelection
+              isCalendarWithVolunteerEvents && !hasCategoryFilterSelection(query)
```

Comment:
i guess `dataVolunteerEvents` should also be ignored in case of location filter selection.

Reply:
Yes, you're right. I'll add it. 👍

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/pull/807/commits/167609db8f6e3f2c14dcb590081f5a57abfa2985